HW45_DB_repository/HW45_DB_repository.py

```diff
@@ -0,0 +1,73 @@
+import sqlite3
+
+
+class Teacher:
+
+    def __init__(self, name, subject, experience_years, salary):
+        self.name = name
+        self.subject = str(subject).capitalize()
+        self.experience_years = experience_years
+        self.salary = salary
+
+    def __repr__(self):
+        return f"Teacher({self.name}, {self.subject}, {self.experience_years}, {self.salary})"
+
+
+class TeachersRepository:
+
+    def __init__(self, db_path):
+        self._connection = sqlite3.connect(db_path, isolation_level=None)
+        self._cursor = self._connection.cursor()
+        self._cursor_execute = self._cursor.execute("CREATE TABLE IF NOT EXISTS Teachers(id INTEGER PRIMARY KEY, name TEXT, subject TEXT, experience_years INTEGER, salary REAL);")
+
+    # adds one record by passed object into methods;
+    def add_teacher(self, teacher: Teacher):
+        return self._cursor.execute(f"INSERT INTO Teachers(name, subject, experience_years, salary) VALUES('{teacher.name}', '{teacher.subject}', {teacher.experience_years}, {teacher.salary});")
+
+    # add several records by passed objects into methods;
+    def add_teachers(self, *teachers):
+        for teacher in teachers:
```

Comment:
Better use execute_many. You use isolation_level=None, so after every insert file will be re-written, so it would be more efficient to perform query for all objects and then save it.